modules/miner/getwork.go

```diff
@@ -0,0 +1,24 @@
+package miner
+
+import (
+	"github.com/NebulousLabs/Sia/modules"
+)
+
+// GetWork() returns a MinerWork struct which can be converted to JSON to be
+// parsed by external miners
+func (m *Miner) GetWork() modules.MinerWork {
+	m.mu.Lock()
+	b := m.blockForWork()
+	target := m.target
+	m.mu.Unlock()
```

Comment:
unless there's a strongly compelling reason, you should have:

```
m.mu.Lock()
defer m.mu.Unlock()
```

at the top of the function. This makes it easier to catch mistakes with locks. For example, if m.blockForWork could return an error, you might forget to unlock before handling the error.  But with the defer, you don't have to remember stuff like that.

It's not a problem in this case but we stick to the conventions anyway.

Generally the only reason that you avoid `defer` on the unlock is if later on in the function you do some sort of network communication or intensive disk io. Everything else tends to be fast enough that it's not worth breaking convention.